meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -0,0 +1,125 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.List;
+
+import org.apache.helix.metaclient.api.MetaClientInterface;
+import org.apache.helix.metaclient.factories.MetaClientConfig;
+
+
+/**
+ * LeaderElectionClient does distributed leader election using CRUD and change notification APIs
+ * provided by underlying metadata client. Leader election config can provide many
+ * configs like base path for all participating nodes, sync/async mode, TTL etc.
+ *
+ * Participants join a leader election group by calling the following API.
+ * The Leader Election client maintains and elect an active leader from participant pool
+ * All participants wanted to be elected as leader joins a pool
+ * LeaderElection client maintains an active leader, by monitoring liveness of current leader and
+ * re-elect if needed and user no need to call elect or re-elect explicitly.
+ * This LeaderElection client will notify registered listeners for any leadership change.
+ */
+public class LeaderElectionClient {
+
+  /**
+   * Construct a LeaderElectionClient using a user passed in leaderElectionConfig. It creates a MetaClient
+   * instance underneath.
+   * When MetaClient is auto closed be cause of being disconnected and auto retry connection timed out, A new
+   * MetaClient instance will be created and keeps retry connection.
+   */
+  public LeaderElectionClient(MetaClientConfig metaClientConfig) {
+
+  }
+
+  /**
+   * Construct a LeaderElectionClient using a user passed in MetaClient object
+   * When MetaClient is auto closed be cause of being disconnected and auto retry connection timed out, user
+   * will need to create a new MetaClient and a new LeaderElectionClient instance.
+   */
+  public LeaderElectionClient(MetaClientInterface MetaClient) {
+
+  }
+
+  /**
+   * Returns true if current participant is able to acquire leadership.
+   */
+  public boolean isLeader(String leaderPath, String participant) {
+    return false;
+  }
+
+  /**
+   * Participants join a leader election group by calling the following API.
+   * The Leader Election client maintains and elect an active leader from participant pool
+   */
+  public boolean joinLeaderElectionParticipantPool(String leaderPath, String participant,
+      Object userInfo) {
+    return false;
+  }
+
+  /**
+   * Any participant may exit the exitLeaderElectionParticipantPool by calling the API.
+   * If the participant is not the current leader, it leaves the pool and won't be elected as next leader.
+   * If the participant is the current leader, it leaves the pool and a new leader will be elected if there are other participants in the pool.
+   * Throws exception if the participant is not in the pool.
+   */
+  public boolean exitLeaderElectionParticipantPool(String leaderPath, String participant) {
```

Comment:
nit: If you want you can rename this to `leave` instead of `exit` cause when you join something you leave 😂